lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -323,6 +324,14 @@ export class UserAgentApplication {
     loginRedirect(userRequest?: AuthenticationParameters): void {
         // validate request
         const request: AuthenticationParameters = RequestUtils.validateRequest(userRequest, true, this.clientId, Constants.interactionTypeRedirect, this.redirectCallbacksSet);
+
+        const requestCorrelationId = userRequest.correlationId || CryptoUtils.createNewGuid();
+        const apiEvent: ApiEvent = new ApiEvent(requestCorrelationId, this.logger);
+        apiEvent.apiEventIdentifier = API_EVENT_IDENTIFIER.LoginRedirect;
+        apiEvent.apiCode = API_CODE.LoginRedirect;
```

Comment:
Should apiEventIdentifier and apiCode be part of the constructor? Looks like they are always set right after initialization?

Reply:
Updated (ish) to be a little cleaner